src/sentry/snuba/metrics/span_attribute_extraction.py

```diff
@@ -126,26 +126,86 @@ def _flatten_query_tokens(conditions: Sequence[QueryToken]) -> list[SearchFilter
     return query_tokens
 
 
-def _parse_conditions(conditions: Sequence[str] | None) -> Sequence[QueryToken]:
-    if not conditions:
-        return []
+# Condition string parsing and transformation
+
+
+def _extend_parsed_condtions(parsed_search_query: Sequence[QueryToken]) -> Sequence[QueryToken]:
+    return _visit_numeric_tokens(parsed_search_query)
+
+
+def _visit_numeric_tokens(parsed_search_query: Sequence[QueryToken]) -> list[QueryToken]:
+    """
+    Visits each token in the parsed search query and converts numeric tokens into paren expressions.
+    """
+    query_tokens: list[QueryToken] = []
+
+    for token in parsed_search_query:
+        if isinstance(token, SearchFilter):
+            query_tokens.append(_extend_numeric_token(token))
+        elif isinstance(token, ParenExpression):
+            query_tokens = query_tokens + _visit_numeric_tokens(token.children)
+        else:
+            query_tokens.append(token)
+
+    return query_tokens
+
+
+def _extend_numeric_token(token: SearchFilter) -> ParenExpression | SearchFilter:
+    """
+    Since all search filter values are parsed as strings by default, we need to make sure that
+    numeric values are treated as such when constructing the rule condition. This function
+    expands the original token into a paren expression if the value is a numeric string.
+
+    Example:
+    `key:'123'` -> `key:'123' OR key:123`
+    `key:['123', '456']` -> `key:['123', '456'] OR key:[123, 456]`
+
+    """
+    if token.operator == "=" or token.operator == "!=":
+        if not str(token.value.value).isdigit():
+            return token
```

Comment:
.isdigit() only returns true if all characters are numeric literals - floats do not fall into this category and are therefore returned as-is - is this on purpose?